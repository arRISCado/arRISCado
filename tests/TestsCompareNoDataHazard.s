.global main
.text

main:
	addi a0, zero, 3
	addi a1, zero, -24
	addi zero, zero, 0
	addi zero, zero, 0
	slt a2, a0, a1
	sltu a3, a0, a1
	slti a4, a0, 2
	slti a5, a0, 4
	slt a6, a1, a0
	sltu a7, a1, a0
	sltiu s2, a0, 1
	sltiu s3, a0, -2

# Expected final state:
# a0: 3
# a1: 4294967272 (-24)
# a2: 0
# a3: 1
# a4: 0
# a5: 1
# a6: 1
# a7: 0
# s2: 0
# s3: 1
