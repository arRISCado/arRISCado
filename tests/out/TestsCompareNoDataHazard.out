0           x0	zero
x           x1	ra
x           x2	sp
x           x3	gp
x           x4	tp
x           x5	t0
x           x6	t1
x           x7	t2
x           x8	s0
x           x9	s1
3           x10	a0
4294967272  x11	a1
0           x12	a2
1           x13	a3
0           x14	a4
1           x15	a5
1           x16	a6
0           x17	a7
0           x18	s2
1           x19	s3
x           x20	s4
x           x21	s5
x           x22	s6
x           x23	s7
x           x24	s8
x           x25	s9
x           x26	s10
x           x27	s11
x           x28	t3
x           x29	t4
x           x30	t5
x           x31	t6